numpy/lib/function_base.py

```diff
@@ -4180,6 +4180,11 @@ def percentile(a,
     if interpolation is not None:
         method = _check_interpolation_as_method(
             method, interpolation, "percentile")
+
+    a = np.asanyarray(a)
+    if a.dtype in [np.csingle, np.cdouble, np.clongdouble]:
```

Comment:
This change breaks for the nan functions. I could do it just for the normal functions, but that makes it inconsistently written. I think it would be best if we don't do the "a.dtype.kind". I'll change the ValueErrors to TypeErrors, though.

Reply:
Done.